purplesmurf1998/Autotracks
# Sprint 6 custom lists

# Description
This PR focuses and the creation and managing of custom vehicle lists. The goal of this functionality is that sales representatives can create a list for their clients, take notes and keep track of the vehicles that interest said client.

Once the alert notification completed, sales reps will receive an alert if a vehicle in one of their lists is sold.
# Acceptance Criteria

- User and CRUD a custom list with a name (client’s name)
- List has date created
- List has date last modified
- User can add and remove vehicles from the list
# How to Test

1. Log in and navigate to the vehicle list page by clicking on the "Vehicle List" navigation button
2. Click on the "Create a custom vehicle list" button to create a new list and write some initial notes (optional)
3. You can remove lists by pressing on the "Bulk Action" dropdown and selecting "Delete"
4. You then need to select the list(s) you wish to delete and press on "Apply", or "Cancel" to abort
5. By clicking on a list, you can view the details as well as the vehicles saved.
6. Pressing "Add vehicles to list" will redirect you to the inventory of the dealership where you will find the "Add to list" option inside the "Select Action" dropdown of the vehicle details page.
7. Deleting vehicles from the list is the same process as deleting a list.
8. The option to save changes done to the notes will appear when changes have been detected
9. To edit the title, click on "Edit" and then "Save"
# Jira Story
[[AP-31] As a user, I would like to manage custom lists for vehicles in the dealership's inventory](https://soen390-team12.atlassian.net/browse/AP-31?atlOrigin=eyJpIjoiNGQzMWU3ODUyYzFlNDc4NmJkZmZkNDFmZTA0Mzc5MWMiLCJwIjoiaiJ9)
# Screenshots
![image](https://user-images.githubusercontent.com/44242944/151480086-7cf51808-08d0-443a-8d31-53c5dc8d451b.png)
![image](https://user-images.githubusercontent.com/44242944/151480163-60b9128e-dd06-42db-8f73-cf464c9c32b0.png)
![image](https://user-images.githubusercontent.com/44242944/151480240-047272d3-2f65-4210-89e4-262d73312d93.png)
![image](https://user-images.githubusercontent.com/44242944/151481764-43d37049-b101-43fc-aedb-5eb7a34f6c5b.png)
![image](https://user-images.githubusercontent.com/44242944/151481832-b6c4d819-3f3e-4948-a04d-a816c4b4af24.png)
![image](https://user-images.githubusercontent.com/44242944/151481812-3bd2fec8-c07a-4110-ba38-a24eb14030d0.png)
![image](https://user-images.githubusercontent.com/44242944/151481895-92e08d48-fbdc-4858-bc73-f3d79d8c3049.png)
![image](https://user-images.githubusercontent.com/44242944/151481938-518e8cc4-0f01-42cd-ac04-56906c8b6d47.png)
